### PR TITLE
Refactor CLI tests

### PR DESCRIPTION
**Pull request by bquenin**
Related to #370 

Note that the tests are not working as they were not working already. I just made some changes to be able to fix them in a subsequent PR.
## Verification

```
$ ./swarm start
$ go test -v github.com/appcelerator/amp/cmd/amp/cli
```

### CODE REVIEW

**Review by freignat91 — APPROVED** [review]
LGTM
Tested ok, expected already bad tests

File: cmd/amp/cli/cli_test.go [hunk 1]
```diff
@@ -1,134 +1,116 @@
...
+	content, err := ioutil.ReadFile(fileName)
 	if err != nil {
-		t.Error(err)
-		return
+		return nil, fmt.Errorf("unable to load test spec: %s. Error: %v", fileName, err)
 	}
-	for _, cmdSpec := range commandMap {
-		cmdString := generateCmdString(cmdSpec)
-		t.Log(cmdString, "Command passed.")
-		for i := 0; i < 10; i++ {
```
Comment (JosephGJ) [inline cmd/amp/cli/cli_test.go:83]: Originally used a small loop to iterate over the command 10 times. The service create command does not block until the service is built in the swarm. 

Tests are currently failing due to this.

```
--- FAIL: TestCmds (0.22s)
        cli_test.go:46: -----------------------------------------------------------------------------------------
        cli_test.go:47: Running spec: test_samples/logs-all.yml
        cli_test.go:97: Running: amp logs
        cli_test.go:46: -----------------------------------------------------------------------------------------
        cli_test.go:47: Running spec: test_samples/logs-metadata.yml
        cli_test.go:97: Running: amp logs  -m
        cli_test.go:46: -----------------------------------------------------------------------------------------
        cli_test.go:47: Running spec: test_samples/logs-numbered.yml
        cli_test.go:97: Running: amp logs  -n 10
        cli_test.go:46: -----------------------------------------------------------------------------------------
        cli_test.go:47: Running spec: test_samples/logs-stack.yml
        cli_test.go:97: Running: amp logs  --stack stack1
        cli_test.go:46: -----------------------------------------------------------------------------------------
        cli_test.go:47: Running spec: test_samples/service-create.yml
        cli_test.go:97: Running: amp service create appcelerator/pinger --name pinger -p www:90:3000
        cli_test.go:46: -----------------------------------------------------------------------------------------
        cli_test.go:47: Running spec: test_samples/service-curl.yml
        cli_test.go:97: Running: curl localhost:90/ping
        cli_test.go:49: miss matched expected output:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed to connect to localhost port 90: Connection refused

FAIL
exit status 1
FAIL    github.com/appcelerator/amp/cmd/amp/cli 0.813s 
```

Comment (bquenin) [reply cmd/amp/cli/cli_test.go:83]: Testing a service creation should probably be in a single test spec (not 2 as it is today), and waiting for the service to be up, included in this test spec.

**Review by subfuzion — APPROVED** [review]
LGTM

**Pull request merged**